packages/smart-account/src/signers/SmartAccountSigner.ts

```diff
@@ -1,47 +1,34 @@
-import { ethers } from 'ethers'
-import { BytesLike, Bytes } from '@ethersproject/bytes'
-import {
-  Web3Provider,
-  ExternalProvider,
-  JsonRpcProvider,
-  Networkish
-} from '@ethersproject/providers'
+import { BigNumber, ethers, Signer as EthersSigner } from 'ethers'
+import { BytesLike } from '@ethersproject/bytes'
+import { JsonRpcProvider } from '@ethersproject/providers'
 import { TypedDataDomain, TypedDataField, TypedDataSigner } from '@ethersproject/abstract-signer'
-import { Signer } from './Signer'
-
-import { Signer as EthersSigner } from '@ethersproject/abstract-signer'
-
-// ChainId , SmartAccountContext, SmartAccountConfig, SmartAccountState from @biconomy-sdk/core-types
-import { ChainId, SendTransactionDto, SignTransactionDto } from '@biconomy-sdk/core-types'
+import { ChainId, SignTransactionDto } from '@biconomy-sdk/core-types'
 
 // Might as well be RpcRelayer
-import { Relayer, RestRelayer } from '@biconomy-sdk/relayer'
-
+// import { Relayer, RestRelayer } from '@biconomy-sdk/relayer'
 import { Deferrable } from 'ethers/lib/utils'
 import { TransactionRequest, TransactionResponse } from '@ethersproject/providers'
 
-// Other ways : // Signer needs config, originalSigner, way to dispatch to rpc-relayer, smart-account-apis
+export class SmartAccountSigner extends EthersSigner implements TypedDataSigner {
+  readonly provider: JsonRpcProvider
+  // readonly sender: JsonRpcSender
+  readonly defaultChainId: number | undefined
 
-export class SmartAccountSigner extends Signer implements TypedDataSigner {
-  // Should be SmartAccountProvider (which makes me want to merge SmartAccountSigner in SmartAccountProvider file)
-  readonly provider: Web3Provider
-  // Review
-  readonly defaultChainId?: number
-
-  constructor(provider: Web3Provider, defaultChainId?: number) {
+  constructor(provider: JsonRpcProvider, defaultChainId?: number) {
     super()
     this.provider = provider
     this.defaultChainId = defaultChainId
+    // this.sender = new JsonRpcSender(provider)
   }
 
   _address!: string
+  // relayer: Relayer
 
   // Might have
   // _context: not smartAccountContext but the addresses of contracts from SmartAccountState
-  //
 
   // TBD
-  private _providers: { [key: number]: Web3Provider } = {}
```

Comment:
Why is this needed?

Reply:
It was there in the code when I started working on this, had removed it.